src/plugins/BunqLayoutPlugin/components/SidebarItem/SidebarItem.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import zenscroll from "zenscroll";
+
+/**
+ * @author Nicola Parrello <nparrello@bunq.com>
+ * @since  20181019 Initial creation.
+ */
+class SidebarItem extends React.Component {
+    /**
+     * @param {Object} props
+     */
+    constructor (props: Object) {
+        super(props);
+
+        const {
+            tag,
+            name,
+            isApiEndpoint
+        } = props;
+
+        this.state = {
+            tag: tag,
+            hash: `#/${tag}`,
+            name: name,
+            isApiEndpoint: isApiEndpoint || false
+        };
+    }
+
+    /**
+     * @param {Event} event
+     */
+    scrollToOperation (event: Event) {
+        let elementSelector;
+
+        event.preventDefault();
+
+        window.history.pushState(null, null, this.state.hash);
+
+        if (this.state.isApiEndpoint) {
+            elementSelector = `#operations-tag-${this.state.tag}`;
+        } else {
+            elementSelector = `#topic-${this.state.tag}`;
+        }
+
+        window.setTimeout(() => {
+            const scrollContainer = document.querySelector(".swagger-ui");
```

Comment:
I think this is fine for this one usecase, but for later you might wanna checkout refs in react and how dom is manipulated, to better stay in the react paradigm and to not directly touch the real dom etc https://reactjs.org/docs/refs-and-the-dom.html#refs-and-finddomnode

Reply:
in this case I went for this implementation because it matches the way Swagger does it, but you are right